File: algorithms/1_easy/N/04024_nearest_available_drone/solution.c
int nearestDrone(int **drones, const int drone_len, int *drones_cols_data, int *target, int target_len) {
    int res_val = INT_MAX;
    int res_pos = -1;
    const int tx = target[0];
    const int ty = target[1];

    for (int i = 0; i < drone_len; i += 1) {
        const int dx = drones[i][0];
        const int dy = drones[i][1];
        const int dr = drones[i][2];

        const int dist = ABS(tx - dx) + ABS(ty - dy);

        if (dist <= dr && dist < res_val) {
            res_val = dist;
            res_pos = i;
        }
    }

    return res_pos;
}

File: algorithms/1_easy/N/04024_nearest_available_drone/solution.rs
impl Solution {
    pub fn nearest_drone(drones: Vec<Vec<i32>>, target: Vec<i32>) -> i32 {
        let mut res: Option<(usize, i32)> = None;
        let (tx, ty) = (target[0], target[1]);

        for (i, d) in drones.iter().enumerate() {
            let (dx, dy, dr) = (d[0], d[1], d[2]);

            let dist = (tx - dx).abs() + (ty - dy).abs();

            if dist <= dr {
                match res {
                    None => {
                        res = Some((i, dist));
                    },
                    Some((_, res_dist)) => {
                        if dist < res_dist {
                            res = Some((i, dist));
                        }
                    }
                }
            }
        }

        match res {
            None => -1,
            Some((idx, _)) => idx.try_into().unwrap(),
        }
    }
}
